mantidproject/mantid
# SANSILL: D11lr and D22lr IPFs

**Description of work.**

Fixes the found issue of parameters used to steer what requires saving in headers of ascii files via `SaveAscii` in DrILL (`log_list_to_save`), and logbook generation (`logbook_default_parameters`, `logbook_optional_parameters`) for low-resolution modes of operation of D11 and D22 instruments at the ILL. In addition, if the log list to save parameter is not set up in a relevant IPF, the empty list of parameters is not going to be forwarded to `SaveAscii`, avoiding errors.

**To test:**

1. Open DrILL
2. Process D11lr data, e.g. 010569 from `Data/UnitTest/D11` making sure `SaveAscii` option it ticked
3. There should be no errors, and saved ascii file(s) should have header(s) containing the following information:
 - facility, 
 - instrument name,
 - experiment identifier,
 - experiment title, 
 - start time,
 - end time,
 - numor list,
 - processedAs flag setting,
 - wavelength,
 - numors with sample transmission source,
 - collimation.actual_position, and 
 - L2


Fixes #32601. 


Release notes added.




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
